Can the ArrayBuilder trait be made Sync?

**Problem**

I encountered an issue when working with async functions and storing a `Box<dyn ArrayBuilder>` (explicitly using `Box<dyn ArrayBuilder + Sync>` solved the issue.

**Proposed solution**

I noticed that currently ArrayBuilder is `ArrayBuilder: Any + Send`, but we can make that into `ArrayBuilder: Any + Send + Sync` can't we? I just tried it locally and `cargo check` didn't give any errors.

**Alternatives**
Manually specifying the `Sync` bound

Hmm I might be confusing then something else, it seems that Vec<Box<...>> is requiring this

```rust
error[E0277]: `(dyn arrow_array::builder::ArrayBuilder + 'static)` cannot be shared between threads safely
   --> src/main.rs:89:5
    |
89  |     tokio::spawn(f);
    |     ^^^^^^^^^^^^ `(dyn arrow_array::builder::ArrayBuilder + 'static)` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `(dyn arrow_array::builder::ArrayBuilder + 'static)`
    = note: required for `Unique<(dyn arrow_array::builder::ArrayBuilder + 'static)>` to implement `Sync`
note: required because it appears within the type `Box<dyn ArrayBuilder>`
   --> /Users/dvic/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/boxed.rs:195:12
    |
195 | pub struct Box<
    |            ^^^
    = note: required for `Unique<Box<(dyn arrow_array::builder::ArrayBuilder + 'static)>>` to implement `Sync`
note: required because it appears within the type `RawVec<Box<dyn ArrayBuilder>>`
   --> /Users/dvic/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:51:19
    |
51  | pub(crate) struct RawVec<T, A: Allocator = Global> {
    |                   ^^^^^^
note: required because it appears within the type `Vec<Box<dyn ArrayBuilder>>`
   --> /Users/dvic/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:396:12
    |
396 | pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
    |            ^^^
note: required because it appears within the type `UpsertsBuilder`
   --> /Users/dvic/Development/qamion/src/ticketmatic.rs:196:8
    |
196 | struct UpsertsBuilder {
    |        ^^^^^^^^^^^^^^
note: required because it appears within the type `TableState<1>`
```

`UpsertsBuilder` contains `Vec<Box<dyn ArrayBuilder>>`

--- a/arrow-array/src/builder/mod.rs
+++ b/arrow-array/src/builder/mod.rs
@@ -232,7 +232,7 @@ use std::any::Any;
 ///     "🍎"
 /// );
 /// ```
-pub trait ArrayBuilder: Any + Send {
+pub trait ArrayBuilder: Any + Send + Sync {
     /// Returns the number of array slots in the builder
     fn len(&self) -> usize;
 

